qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/page/VisitNotePage.java

```diff
@@ -20,7 +20,7 @@ public class VisitNotePage extends Page {
 	private static final By CODE = By.className("code");
 	private static final By UI_ID_1 = By.id("ui-id-1");
 	private static final By UI_MENU_ITEM = By.className("ui-menu-item");
-	private static final By PRIMARY_DIAGNOSIS_ELEMENT = By.cssSelector(".diagnosis.primary .matched-name");
```

Comment:
Isn't the cssSelector a better element locator than xpath?

Reply:
Using xpath makes it to fail.

Reply:
> Using xpath makes it to fail.

Why not leave `cssSelector` then?

Reply:
Sorry i meant that using xpath make it work perfect

Reply:
The cssSelector has been passing before and is a better locator than xpath, could you be know why it's now failing in github actions and yet passing in CI?

Reply:
True, I guess its due to the intensity of the server, Or something like heavy loading of the files.

Reply:
@sherrif10 is that reason enough to change it from `cssSelector`?

Reply:
Fairly enough, Using XPath instead made it work without loading issues.

Reply:
@sherrif10 can confirm that reverting it to cssSelector makes it fail?

Reply:
Reverted it back.

Reply:
@dkayiwa  Hope this is comfirming for the visit note test., You might say why not adding waiting element, though adding waiting element doesnt resolve it , using xpath resolves better without adding waiting element which i had initially did.

Reply:
I do not believe it unless you prove it.

Reply:
Great, so i think you can prove it after using CSS selector. still issue  occurs with visibility of an element, Actually thats why i added those waiting elements which didnt add moreof value.

Reply:
Why is it passing on Bamboo?

Reply:
Look for a solution which uses `cssSelector`

Reply:
@dkayiwa . Using cssSelector breaks the workflow with  visibility of an element issues as before, i tried all cssSelector methods but still breaks, the reason is, the primary and secondary bars are not attached to the page where CSS selector would pick the pre-incident element correctly, Using XPath picks the exact element which resolves the issue. Am open to recommendations for using` cssSelector than xpath`

Reply:
Do you mind explaining why, even with your change to xpath, the firefox tests are still failing?

Reply:
Can you do so?

Reply:
@dkayiwa, The build is happy now.